client/observeResource.go

```diff
@@ -214,12 +214,12 @@ func (c *Client) ObserveResource(
 		return "", err
 	}
 
-	err = c.deviceCache.StoreDeviceToPermanentCache(d)
-	if err != nil {
-		return "", err
+	refDev, stored := c.deviceCache.TryStoreDeviceToPermanentCache(d)
+	if !stored {
+		d.Release(ctx)
 	}
 
-	d.Acquire()
+	refDev.Acquire()
```

Comment:
this can lead to memory leak. - `.TryStoreDeviceToPermanentCache` returns already incremented ref counter.